sdk/src/main/java/io/opentelemetry/sdk/trace/SpanWrapper.java

```diff
@@ -30,6 +30,17 @@
 import java.util.List;
 import javax.annotation.concurrent.Immutable;
 
+/**
+ * Immutable class that stores {@link SpanData} based on a {@link RecordEventsReadableSpan}.
+ *
+ * <p>This is a variant of the "flyweight" design pattern, to reduce overhead. We store a reference
```

Comment:
I don't think this is actually a flyweight in the GoF sense, which is more like a multi-singleton. This is a pure delegate.